src/vm.rs

```diff
@@ -172,17 +259,17 @@ impl<VCpuType: VirtualCPU> UhyveVm<VCpuType> {
 			unsafe { self.mem.as_slice_mut() } // slice only lives during this fn call
 				.try_into()
 				.expect("Guest memory is not large enough for pagetables"),
+				self.mem.guest_address.as_u64()
 		);
 	}
 
 	pub fn load_kernel(&mut self) -> LoadKernelResult<()> {
+		// TODO: Remove the duplicate load in load_kernel.
 		let elf = fs::read(self.kernel_path())?;
 		let object = KernelObject::parse(&elf).map_err(LoadKernelError::ParseKernelError)?;
 
-		// TODO: should be a random start address, if we have a relocatable executable
-		let kernel_start_address = object.start_addr().unwrap_or(0x400000) as usize;
-		let kernel_end_address = kernel_start_address + object.mem_size();
-		self.offset = kernel_start_address as u64;
+		let kernel_end_address = self.start_address.as_u64() as usize + object.mem_size();
+		self.offset = self.start_address.as_u64();
```

Comment:
Currently, UhyveVm has two entries: `self.offset`, `self.start_address`, which effectively do the same thing. One of them should be removed, probably.